### DISCUSSION TITLE
Unable to acknowledge alerts; throws "Request failed with status code 400" error

### DISCUSSION THREAD

**Discussion by cyb3rz3us** [Security-Onion-Solutions/securityonion#7631] [Unsupported Versions] [answered]
### General

**SO version:** 2.3.70
**Platform:** VM within ESX 6.5
**vCPU:** 8
**RAM:** 24GB
**HDD:** system = 500GB; nsm = 1TB
**RAID status:** not implemented
**Installed using:** SO ISO

### Description

Up through 03/22/22, I was able to acknowledge alerts with no issues and then at some point in the early morning of 03/23/22, my system experienced an ABEND (ABnormal END); a completely unintentional and uninitiated hard reset. After a reboot and SO restart, I am no longer able to acknowledge alerts; when attempted, I receive the _"Request failed with status code 400"_ error.

The board is green in that repeated ```sudo so-status``` commands return all OKs.

I checked the `/opt/so/log/soc/sensoroni-server.log` file and each time I attempt an acknowledge, the following error is seen (message abbreviated to avoid posting a ton of repeating data):
```
timestamp=2022-03-23T20:35:07.011672158Z level=error message="Encountered error while updating elasticsearch" clientHost=https://192.168.5.247:9200 error="search_phase_execution_exception:  -> {\n  \"error\" : {\n    \"root_cause\" : [ ],\n    \"type\" : \"search_phase_execution_exception\",\n    \"reason\" : \"\",\n    \"phase\" : \"can_match\",\n    \"grouped\" : true,\n    \"failed_shards\" : [ ],\n    \"caused_by\" : {\n      \"type\" : \"search_phase_execution_exception\",\n      \"reason\" : \"Search rejected due to missing shards
```

In addition, there are hundreds of these messages coming in at about 1\second:
```
timestamp=2022-03-23T19:52:18.319322549Z level=warn message="Host not found in raid status metrics" host=tyr
timestamp=2022-03-23T19:52:18.319925021Z level=info message="Handled request" contentLength=20 elapsedMs=0 impl=*server.NodeHandler method=POST path=/api/node remoteAddr=172.17.0.1:56694 requestId=6b006bab-2d83-49df-92b8-799f5f208c63 requestor=SONODE sourceIp=192.168.5.247 statusCode=200
```

Not sure what else to check or if there are some basic corrective measures\cleanup one should perform following such an event.

Thanks in advance for your help...

**Comment by cyb3rz3us** [comment]
Brief update - restarted elasticsearch and see a few ```failed to recover from translog``` errors as well as lots of failed shards so it seems the translog got corrupted which makes sense given the nature of an ABEND.  It looks like I will need to truncate the translog using the ```elasticsearch-shard``` tool however, the only place I find it is in the ```so-elasticsearch``` container.

But here's the rub - this tool cannot be run while elasticsearch is running.  After stopping the container, the tool is accessible from outside via a docker overlay directory but I'm pretty sure the translog is inside the container and becuase I'm not sure if the translog is also accessible outside of the container, will this even work and if so, is it safe?

Any insights here are appreciated...

**Comment by cyb3rz3us** [comment]
OK - this appears to be **partially resolved**.  I say partially because while I have corrected the index corruption, I now have an issue where whenever ES is restarted, the node health goes to red which restarts the self-healing process; this occurs even though the **node health was green prior to the restart**.  The process takes roughly 2 hours during which indices are restored & shards brought online all resulting in the node health going to green.  Regardless (and perhaps I'm wrong here), but I don't think this is normal ES behavior even in a standalone environment.

I will admit that this could be an ES issue but I figured it is worth a ping to the SO community to see if there are any ideas as to why this is occurring.

**Comment by dougburks** [comment]
Are there files left over in `/nsm/elasticsearch/` that are causing Elasticsearch to think it needs to restart the self-healing process?

**Reply by cyb3rz3us** [reply]
There are no files in ```/nsm/elasticsearch``` but there are some directories.  I have attached the abbreviated tree output below.  Also, tons of files in the ```/nsm/elasticsearch/nodes/0/_state``` directory.

```
/nsm/elasticsearch/
├── nodes
│   └── 0
│       ├── indices
│       ├── node.lock
│       ├── snapshot_cache
│       └── _state
└── repo
```

**Reply by dougburks** [reply]
Take a look at https://discuss.elastic.co/t/es-failed-to-recover-after-crash/8195/4 and see if you might have some remaining translog files that are causing Elasticsearch to restart the self-healing process.

**Reply by cyb3rz3us** [reply]
Thanks @dougburks - that thread discusses getting a node back to green from a red condition but not really why a green node goes to red immediately after restart...still digging...

**Reply by dougburks** [reply]
My interpretation of that thread was that that they were indeed discussing why a green node goes to red immediately after restart. From the thread:
> In nel's case it was corrupted transaction log. When you run out of disk
space sometimes the last transaction cannot be fully written into
transaction log and then it fails on recovery.

**Comment by cyb3rz3us** [comment, marked as answer]
"...and then it fails on recovery..." --- True but I haven't had any issues with recovery failures or disk space.

That aside, in discussions with ES, I have learned that in an SO standalone install, it is normal - even expected - that a node will be red after startup and will take some amount of time for it to move into green. How long this takes depends largely on how the quantity and size of the indices\shards. If one wants a node to be green shortly (or even immediately) after startup, then apparently one would need to implement SB shards which is something I will validate in the future.

Thanks again.
